exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -1851,9 +1922,73 @@ private static String relativize(String baseDir, String childPath) {
           .relativize(fullPathWithoutSchemeAndAuthority.toUri()));
       if (relativeFilePath.isAbsolute()) {
         throw new IllegalStateException(String.format("Path %s is not a subpath of %s.",
-            basePathWithoutSchemeAndAuthority.toUri().toString(), fullPathWithoutSchemeAndAuthority.toUri().toString()));
+            basePathWithoutSchemeAndAuthority.toUri().getPath(), fullPathWithoutSchemeAndAuthority.toUri().getPath()));
+      }
+      return relativeFilePath.toUri().getPath();
+    }
+  }
+
+  /**
+   * Supported metadata versions.
+   * <p>
+   * Note: keep them synchronized with {@link ParquetTableMetadataBase} versions
+   */
+  public static class MetadataVersion {
```

Comment:
1. Since this class is used in  other classes, I suggest we factor out as separate class. This class has too many inner classes.
2. It's not an enum, in this case name should be plural - `MetadataVersions`.